ntbs-service-unit-tests/Jobs/UpdateTableCountsJobTest.cs

```diff
@@ -81,92 +95,67 @@ public async Task Run_LogsInfo_WhenNoFieldsReduced(int count)
-                LTBR_DiseasePeriodCount = ltbrDiseasePeriodCount,
-                LTBR_PatientEpisodeCount = ltbrPatientEpisodeCount,
-                NotificationClusterMatchCount = notificationClusterMatchCount,
-                NotificationSpecimenMatchCount = notificationSpecimenMatchCount,
-                EtsSpecimenMatchCount = etsSpecimenMatchCount
-            };
+            var previousTableCounts = CreateBasicTableCounts(0);
+            typeof(TableCounts)
+                .GetProperty(fieldName)
+                .SetValue(previousTableCounts, 1);
 
-            var tableCounts = new[] { CreateBasicTableCounts(), previousTableCounts };
+            var tableCounts = new[] { CreateBasicTableCounts(0), previousTableCounts };
             _tableCountsRepository
                 .Setup(r => r.GetRecentTableCounts())
                 .Returns(Task.FromResult(tableCounts.AsEnumerable()));
 
-            Log.Logger = new LoggerConfiguration().WriteTo.TestCorrelator().CreateLogger();
-
-            using (TestCorrelator.CreateContext())
-            {
-                // Act
-                await _updateTableCountsJob.Run(null);
+            // Act and assert
+            var exception = await Assert.ThrowsAsync<ApplicationException>(() => _updateTableCountsJob.Run(null));
+            Assert.Contains("Some table counts have decreased", exception.Message);
+            Assert.Contains($"Property: {fieldName} has decreased from 1 to 0", exception.Message);
+        }
 
-                // Assert
-                var logEvents = TestCorrelator.GetLogEventsFromCurrentContext().ToList();
+        [Fact]
+        public async Task Run_LogsWarnings_WhenAllFieldsAreReduced()
+        {
+            // Arrange
+            var tableCounts = new[] { CreateBasicTableCounts(0), CreateBasicTableCounts(1) };
+            _tableCountsRepository
+                .Setup(r => r.GetRecentTableCounts())
+                .Returns(Task.FromResult(tableCounts.AsEnumerable()));
 
-                Assert.Contains(logEvents, logEvent => logEvent.Level > LogEventLevel.Information);
-                Assert.DoesNotContain(logEvents, logEvent => logEvent.Level > LogEventLevel.Warning);
-                Assert.Contains(logEvents, logEvent =>
-                    logEvent.Level == LogEventLevel.Warning
-                    && logEvent.RenderMessage().Contains("Some table counts have decreased"));
-            }
+            // Act and assert
+            var exception = await Assert.ThrowsAsync<ApplicationException>(() => _updateTableCountsJob.Run(null));
+            Assert.Contains("Some table counts have decreased", exception.Message);
+            CountNames.ForEach(fieldName =>
+                Assert.Contains($"Property: {fieldName} has decreased from 1 to 0", exception.Message)
+            );
         }
 
+        private static IEnumerable<string> CountNames => new[]
+        {
+            "MigrationNotificationsViewCount",
```

Comment:
I think you can make these a bit safer by replacing with e.g. `nameof(TableCounts.MigrationNotificationsViewCount)`

Reply:
I didn't know that was a thing, thanks! Done